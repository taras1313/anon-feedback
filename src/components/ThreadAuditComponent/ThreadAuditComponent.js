import React from 'react';
import cx from 'classnames';

import ThumbDownOutlinedIcon from '@material-ui/icons/ThumbDown';
import ThumbUpOutlinedIcon from '@material-ui/icons/ThumbUp';

import styles from './ThreadAuditComponent.module.scss';

export const ThreadAuditComponent = ({
	subscribersCount,
	dislikesCount,
	likesCount,
	isClickable,
	like,
	dislike,
	likeStatus,
	createdDate
}) => {
	return (
		<div className={styles.threadMetaInfo}>
			{subscribersCount} subscribers
			<span className={styles.divider}>-</span>
			{createdDate}
			<span className={styles.divider}>-</span>
			<div onClick={isClickable ? () => like() : null} className={styles.flex}>
				<ThumbUpOutlinedIcon
					className={cx(styles.icon, {
						[styles.like]: likeStatus === 'liked'
					})}
				/>{' '}
				{likesCount}{' '}
			</div>
			<span className={styles.divider}>-</span>
			<div onClick={isClickable ? () => dislike() : null} className={styles.flex}>
				<ThumbDownOutlinedIcon
					className={cx(styles.icon, {
						[styles.dislike]: likeStatus === 'disliked'
					})}
				/>{' '}
				{dislikesCount}{' '}
			</div>
		</div>
	);
};
