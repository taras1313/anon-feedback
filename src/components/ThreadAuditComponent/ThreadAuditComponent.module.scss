@import "../../styles/variables";

.divider {
  margin: 0 8px;
}

.icon {
  max-width: 20px;
  max-height: 20px;
  margin-right: 6px;
}

.threadMetaInfo {
  color: $light-gray;
  display: flex;
  font-size: 13px;
  width: 100%;
  align-items: flex-end;
}