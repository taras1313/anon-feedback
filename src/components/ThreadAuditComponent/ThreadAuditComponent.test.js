import React from 'react';
import { ThreadAuditComponent } from './ThreadAuditComponent';
import { shallow } from 'enzyme';

describe('threadAuditComponent', () => {
	test('renders w/o props', () => {
		const wrapper = shallow(<ThreadAuditComponent />);

		expect(wrapper).toMatchSnapshot();
	});

	test('like and dislike work as expected', () => {
		const like = jest.fn();
		const dislike = jest.fn();
		const wrapper = shallow(<ThreadAuditComponent like={like} dislike={dislike} isClickable />);

		wrapper.find("[data-testid='like-button']").simulate('click');
		expect(like).toHaveBeenCalled();

		wrapper.find("[data-testid='dislike-button']").simulate('click');
		expect(dislike).toHaveBeenCalled();
	});
});
