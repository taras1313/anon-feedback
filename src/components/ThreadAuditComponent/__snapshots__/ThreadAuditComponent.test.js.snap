// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`threadAuditComponent renders w/o props 1`] = `
<div
  className="threadMetaInfo"
>
   subscribers
  <span
    className="divider"
  >
    -
  </span>
  <span
    className="divider"
  >
    -
  </span>
  <div
    className="flex"
    data-testid="like-button"
    onClick={null}
  >
    <ThumbUpIcon
      className="icon"
    />
     
     
  </div>
  <span
    className="divider"
  >
    -
  </span>
  <div
    className="flex"
    data-testid="dislike-button"
    onClick={null}
  >
    <ThumbDownIcon
      className="icon"
    />
     
     
  </div>
</div>
`;
